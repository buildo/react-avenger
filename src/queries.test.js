import * as React from 'react';
import declareQueries from './queries';
import { Query } from 'avenger';
import * as t from 'io-ts';
import { mount } from 'enzyme';

const makeQueries = () => {
  const foo = Query({
    params: { token: t.string },
    fetch: ({ token }) => Promise.resolve(token)
  });
  return { foo };
};

const sleep = (timeout) => new Promise(resolve => {
  setTimeout(resolve, timeout);
});

describe('declareQueries', () => {

  it('works', async () => {
    const { foo } = makeQueries();
    const render = jest.fn(JSON.stringify.bind(JSON));
    const WithFoo = declareQueries({ foo })(render);
    const mounted = mount(<WithFoo token='foo' />);
    expect(mounted).toMatchSnapshot();
    await sleep(10);
    expect(render.mock.calls.length).toBe(2);
    expect(render.mock.calls[0][0].readyState.foo).toEqual({ ready: false, loading: true });
    expect(render.mock.calls[0][0].foo).toBe(undefined);
    expect(render.mock.calls[1][0].readyState.foo).toEqual({ ready: true, loading: false });
    expect(render.mock.calls[1][0].foo).toBe('foo');
  });

});
