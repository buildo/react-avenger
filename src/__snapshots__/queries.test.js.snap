// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`declareQueries works 1`] = `
<queries(mockConstructor)
  token="foo"
>
  <mockConstructor
    foo={
      Object {
        "loading": true,
        "ready": false,
      }
    }
    token="foo"
  >
    {"token":"foo","foo":{"loading":true,"ready":false}}
  </mockConstructor>
</queries(mockConstructor)>
`;

exports[`declareQueries works with querySync 1`] = `
<queries(mockConstructor)
  token="foo"
>
  <mockConstructor
    foo={
      Object {
        "loading": false,
        "ready": true,
        "value": "foo",
      }
    }
    token="foo"
  >
    {"token":"foo","foo":{"ready":true,"loading":false,"value":"foo"}}
  </mockConstructor>
</queries(mockConstructor)>
`;
